src/index.js

```diff
@@ -0,0 +1,276 @@
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(".modal__close-btn");
+
+// Card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+// Change avatar functionality
+const changeAvatarButton = document.querySelector(".profile__avatar-btn");
+const changeAvatarModal = document.querySelector("#edit-avatar-modal");
+const changeAvatarForm = changeAvatarModal.querySelector("#edit-avatar-form");
+const avatarSubmitButton =
+  changeAvatarModal.querySelector(".modal__submit-btn");
+const changeAvatarClose = changeAvatarModal.querySelector(".modal__close-btn");
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d44cd1da-3eb5-4df7-b9ca-8d742a5da7d3",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+    const profileAvatar = document.querySelector(".profile__avatar");
+    profileAvatar.src = userInfo.avatar;
+
+    cards.forEach((item) => {
+      const cardEl = getCardElement(item);
+      cardsList.append(cardEl);
+    });
+  })
+  .catch(console.error);
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  addEscapeListener(); // Add
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  removeEscapeListener(); // Remove
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  editSubmitButton.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      editSubmitButton.textContent = "Save";
+    })
+    .catch(console.error);
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  cardSubmitButton.textContent = "Saving...";
+  api
+    .addNewCard(inputValues)
+    .then((card) => {
+      const cardElement = getCardElement(card);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      cardForm.reset();
+      disableButton(evt.submitter, settings);
+      cardSubmitButton.textContent = "Save";
+    })
+    .catch(console.error);
+}
+
+changeAvatarForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  const formData = new FormData(event.target);
+  avatarSubmitButton.textContent = "Saving...";
+  // output as an object
+  const data = Object.fromEntries(formData);
+  api
+    .editAvatar(data)
+    .then(() => {
+      const avatar = document.querySelector(".profile__avatar");
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine